Migrate to core kit

arnaldo2792 opened the pull request:
**Issue number:**

Related to #4048

**Description of changes:**

This PR migrates the code to use the Bottlerocket core kit.


**Testing done:**
- Built an AMI with my personal core kit, and verified the host booted.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

Commit:
migrations: relocate to settings-migrations

Migrations need to stay in-tree along with plugins and defaults,
because they are directly related to how settings are configured and
how they evolve from one release to the next.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
add settings-migrations package

Signed-off-by: Ben Cressey <bcressey@amazon.com>

arnaldo2792 committed:
packages: remove unnecessary packages

They are now provided by the Bottlerocket Core Kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
sources: remove unnecessary sources

They were migrated to the Bottlerocket Core Kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
variants: add settings-plugins and settings-extensions as build dependencies

All the packages the variants depend on are already built in the
Bottlerocket Core Kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
docs: remove GLOSSARY.md

It was migrated to the Bottlerocket core kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
Makefile: update for compatibility with Bottlerocket Core Kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
workspace: migrate to Cargo.toml

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
core-kit: use v2.0.0

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
sources: fix clippy warnings

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

arnaldo2792 committed:
github: adapt github actions for migration to core kit

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

cbgbt reviewed (commented):
Nice! :rocket: 

This looks good to me so far, I'll take another look when the tests are passing.

sam-berning commented on sources/Cargo.toml:
```diff
@@ -1,26 +1,9 @@
-    "api/thar-be-updates",
-    "api/settings-committer",
-    "api/migration/migrator",
+    "api/apiclient",
```
nit: this is out of alphabetical order with datastore

sam-berning commented on sources/Cargo.toml:
```diff
@@ -113,22 +69,7 @@ members = [
-    "updater/update_metadata",
-    "updater/updog",
-
     "constants",
```
duplicate?

sam-berning commented on variants/aws-ecs-1-nvidia/Cargo.toml:
```diff
@@ -39,3 +39,4 @@ path = "../variants.rs"
 [build-dependencies]
 settings-defaults = { path = "../../packages/settings-defaults" }
 settings-plugins = { path = "../../packages/settings-plugins" }
+settings-migrations = { path = "../../packages/settings-migrations" }
```
nit: out of alphabetical order

bcressey commented:
Looks good to me once we have the first core kit release.

Commit:
settings-plugins: do not prefer shared linking

The SDK is built for two host architectures - x86_64 and aarch64.
Each build includes the same two target architectures. However, the
libstd.so binary for a given target architecture will not be the same
across the two host SDK images, because they are built on different
systems and at different times.

To reduce file sizes, we prefer dynamically linking to libstd for all
Rust binaries, and arrange to install libstd through the libstd-rust
package. Prior to the kit migration, we always built the final image
with the same host SDK image that was used to build all packages, so
the question of which libstd should be shipped never came up.

In the kits world, we need the libstd-rust that corresponds to the
many Rust binaries that come from the core kit. However, if we prefer
dynamic linking, then we also need the libstd-rust that matches the
Rust binaries built locally, which may not be the same if the kit was
built on a different host architecture.

Sidestep this for now by not preferring shared linking for settings
plugins, so that they are statically linked to the libstd present in
the build environment, and do not require it to be installed in the
final image.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

cbgbt reviewed (approved):
:rocket:
